tests/test_api.py

```diff
@@ -161,7 +166,7 @@ async def test_open_array(memory_store: MemoryStore, zarr_format: ZarrFormat) ->
     assert z.read_only
 
     # path not found
-    with pytest.raises(FileNotFoundError):
+    with pytest.raises(ArrayNotFoundError):
```

Comment:
This is currently failing the tests, because it actually raises a `GroupNotFoundError` now. I think the sensible thing to do here would be to create another new error, `NodeNotFoundError`, that is raised when a non-specific function is used (e.g., just `open`) to try and open either an array or a group.